### PR TITLE
Consolidated Reservoir Class and Linear Outlet

### PR DESCRIPTION
**Pull request by jdmattern-noaa**
Nonlinear and linear reservoir classes consolidated into one reservoir class. Reservoir linear outlet class added.

## Additions
Reservoir linear outlet class inherited from reservoir outlet class.

## Removals
Removed linear reservoir class and changed name of nonlinear reservoir class to just reservoir.

## Changes
Changed inherited names throughout different models from nonlinear reservoir to reservoir.

## Testing

1. Passes hymod, t-shirt, and unit tests. Modified one reservoir unit test to account for linear outlet.

## Notes
Reservoir linear outlet class inherits from outlet class and sets the exponent "b" to 1 at construction to create linearity in the outlet velocity function used from the outlet class. Should the linear class instead override the outlet velocity function with a modified equation that does not have the exponent "b" to create a linear equation?

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows project standards (link if applicable)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: src/models/kernels/reservoir/Reservoir_Linear_Outlet.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @brief Parameterized Constructor that builds a Linear Reservoir Outlet object.
+ * @param a outlet velocity calculation coefficient
+ * @param activation_threshold_meters meters from the bottom of the reservoir to the bottom of the outlet
+ * @param max_velocity_meters_per_second max outlet velocity in meters per second
+ */
+Reservoir_Linear_Outlet::Reservoir_Linear_Outlet(double a, double activation_threshold_meters,
+                             double max_velocity_meters_per_second) :
+        Reservoir_Outlet(a, 1.0, activation_threshold_meters, max_velocity_meters_per_second)
```
Comment (jdmattern-noaa) [inline src/models/kernels/reservoir/Reservoir_Linear_Outlet.cpp:19]: Reservoir linear outlet class inherits from outlet class and sets the exponent "b" to 1 at construction to create linearity in the outlet velocity function used from the outlet class. Should the linear class instead override the outlet velocity function with a modified equation that does not have the exponent "b" to create a linear equation?

Comment (hellkite500) [reply src/models/kernels/reservoir/Reservoir_Linear_Outlet.cpp:19]: Might not be a bad idea just to show it explicitly.

File: test/models/hymod/include/Reservoir_Test.cpp [hunk 2]
```diff
@@ -55,47 +62,63 @@ class NonlinearReservoirKernelTest : public ::testing::Test {
...
-    setupMultipleOutletOutOfOrderNonlinearReservoir();
+    setupMultipleOutletOutOfOrderReservoir();
 
     setupExponentialOutletReservoir();
 }
 
-void NonlinearReservoirKernelTest::TearDown() {
+void ReservoirKernelTest::TearDown() {
 
 }
```
Comment (hellkite500) [inline test/models/hymod/include/Reservoir_Test.cpp:85]: Should add some basic test coverage for the add outlet function that was added.  Should be able to add an outlet and assert the number of outlets and the sorted order of activation thresholds

Comment (jdmattern-noaa) [reply test/models/hymod/include/Reservoir_Test.cpp:85]: @jdmattern-noaa Adding GitHub issue to add more Reservoir unit tests to directly test particular functions in upcoming PRs as discussed with @hellkite500.

**Pull request merged**